langmeta/langmeta/shared/src/main/scala/org/langmeta/semanticdb/internal/package.scala

```diff
@@ -10,132 +10,192 @@ import org.langmeta.inputs.{Input => dInput}
+                    val dendOffset = dinput.lineToOffset(srange.endLine) + srange.endCharacter
+                    val ddefnpos = dPosition.Range(dinput, dstartOffset, dendOffset)
+                    d.ResolvedName(ddefnpos, dsym, disDefinition)
+                  case other =>
+                    sys.error(s"bad protobuf: unsupported occurrence $other")
+                }.toList
+              }.getOrElse(Nil)
+            }
             val dmembers: List[d.Signature] = smembers.toIterator.map { smember =>
               if (smember.endsWith("#")) d.Signature.Type(smember.stripSuffix("#"))
               else if (smember.endsWith(".")) d.Signature.Term(smember.stripSuffix("."))
               else sys.error(s"Unexpected signature $smember")
             }.toList
             val doverrides = soverrides.flatMap(d.Symbol.unapply).toList
-            val ddefn = d.Denotation(dflags, dname, dsignature, dnames, dmembers, doverrides)
-            Some(d.ResolvedSymbol(dsym, ddefn))
-          case other => sys.error(s"bad protobuf: unsupported denotation $other")
+            Some(d.ResolvedSymbol(dsym, d.Denotation(dflags, dname, dsignature, dnames, dmembers, doverrides)))
+          case other => sys.error(s"bad protobuf: unsupported symbol information $other")
         }
       }
       object sSynthetic {
         def unapply(ssynthetic: s.Synthetic): Option[dSynthetic] = ssynthetic match {
-          case s.Synthetic(Some(sPosition(dpos)), dtext, snames) =>
-            val dnames = snames.toIterator.map {
-              case s.ResolvedName(Some(s.Position(sstart, send)), d.Symbol(dsym), disDefinition) =>
-                val dsyntheticinput = dInput.Synthetic(dtext, dpos.input, dpos.start, dpos.end)
-                val dsyntheticpos = dPosition.Range(dsyntheticinput, sstart, send)
-                d.ResolvedName(dsyntheticpos, dsym, disDefinition)
-              case other =>
-                sys.error(s"bad protobuf: unsupported name $other")
-            }.toList
+          case s.Synthetic(Some(sRange(dpos)), stext) =>
+            val dtext = stext.map(_.text).getOrElse("")
+            val dnames = {
+              stext.map { stext =>
+                stext.occurrences.toIterator.map {
+                  case s.SymbolOccurrence(Some(srange), d.Symbol(dsym), sRole(disDefinition)) =>
+                    val dsyntheticinput = dInput.Synthetic(dtext, dpos.input, dpos.start, dpos.end)
+                    val dstartOffset = dsyntheticinput.lineToOffset(srange.startLine) + srange.startCharacter
+                    val dendOffset = dsyntheticinput.lineToOffset(srange.endLine) + srange.endCharacter
+                    val dsyntheticpos = dPosition.Range(dsyntheticinput, dstartOffset, dendOffset)
+                    d.ResolvedName(dsyntheticpos, dsym, disDefinition)
+                  case other =>
+                    sys.error(s"bad protobuf: unsupported occurrence $other")
```

Comment:
General note, I've found the error messages in these conversion to be cryptic. If for example `d.Symbol` fails then I want the error message from that apply instead of a generic "bad protobuf". Pattern matching is not helping us here IMO.